python/lsst/analysis/tools/actions/scalar/scalarActions.py

```diff
@@ -284,3 +288,107 @@ def __call__(self, data: KeyedData, **kwargs) -> Scalar:
             return 100.0 * result
         else:
             return result
+
+
+class DivideScalar(ScalarAction):
+    """Calculate (A/B) for scalars."""
+
+    actionA = ConfigurableActionField[ScalarAction](doc="Action which supplies scalar A")
+    actionB = ConfigurableActionField[ScalarAction](doc="Action which supplies scalar B")
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        yield from self.actionA.getInputSchema()
+        yield from self.actionB.getInputSchema()
+
+    def __call__(self, data: KeyedData, **kwargs) -> Scalar:
+        """Return the result of A/B.
+
+        Parameters
+        ----------
+        data : `KeyedData`
+
+        Returns
+        -------
+        result : `Scalar`
+            The result of dividing A by B.
+        """
+        scalarA = self.actionA(data, **kwargs)
+        scalarB = self.actionB(data, **kwargs)
+        if scalarB == 0:
+            raise ValueError("Denominator is zero!")
+        return scalarA / scalarB
+
+
+class CountThreshold(ScalarAction):
```

Comment:
I was working on incorporating all your suggestions in the code, but then stumbled upon this issue- threshold only accepts float values, so passing "nan" as a value causes an exception. There are two ways I see if we wanna continue moving forward with this approach-
1. Have some default float value passed that represents "nan"- doesn't make much sense, but a potential solution nonetheless.
2. Have another parameter, called "nan" that governs if NaNs should be part of the counting. I have put together some code using this approach and would like your thoughts on it.
Of course, if you have any other potential solution, I'm happy to include it.

Reply:
I think if you make the pex_config field type `optional = True` and not give it a default value, then in the code accessing the field will resolve to `None`. i.e. `if self.threshold is None`. This could function as the sentinal you were talking about and you can take None to mean threshold on NaN.

Reply:
I tried your suggestion for the following three scenarios-
1. Count just the NaNs - works if threshold = None and op = "eq"
2. Count just the non-NaNs - works if threshold = None and op = "ne"
3. Count NaNs as well as non-NaNs - doesn't work since I couldn't think of a way to tell the code to count NaNs and non-NaNs just with those two parameters. We could leave this on the user to calculate by adding NaN and non-NaN counts, but it seems a basic metric that should already be baked in the code.

Maybe you could go through the code that I've already pushed and see if that makes sense? It already works for all the above 3 scenarios and other edge cases.

Reply:
I'm just a bit confused, did you push the solution that allows eq, or are you saying you want me to review what is here first?

Reply:
Also I'm not sure I understand your point 3 re-reading it? In what scenario would a metric could both nans and non nans in the same configured action? Are you saying you want a combination where this tool just outputs the length of the catalog?

Reply:
I tried your suggestions (importing math/math.nan in my module and passing math.nan/nan in the yaml) and it doesn't work. The module resolution doesn't work in either cases. It still complains that I passed string (nan/math.nan) instead of float.

Reply:
I was trying a couple of things with the YAML file and this is what worked-
`atools.numDiaSourcesNanReliability.process.calculateActions.countingAction.threshold: !!float nan` 
Aadding a `!!float` before `nan` allows for successful module resolution and then correct calculation of metrics. Should I go ahead with this implementation?

Reply:
I've pushed those changes for you to go through. Turns out adding `!!<tag>` is a standard method to explicitly specify the data type in the YAML. I also got rid of the NumDiaSourcesAllMetric function since it was using a version of the new NumDiaSourcesSelectionMetric function. Please let me know in case you have any concerns.